### PR TITLE
On-chain data - deviation trigger feeds

### PR DESCRIPTION
**Pull request by DZariusz**

### CODE REVIEW

File: src/factories/PriceDataFactory.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import Leaf from '../types/Leaf';
+import {DeviationFeeds, DeviationFeed, PriceData, PriceDataByKey} from '../types/DeviationFeeds';
+
+export class PriceDataFactory {
```
Comment (DZariusz) [inline src/factories/PriceDataFactory.ts:4]: @leossantiago this can be QAed

File: src/services/tools/LeavesToRecord.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import Leaf from "../../types/Leaf";
+
+export class LeavesToRecord {
```
Comment (DZariusz) [inline src/services/tools/LeavesToRecord.ts:3]: @leossantiago this can be QAed

File: src/services/deviationsFeeds/PriceDataOverflowChecker.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    const priceMaxUint128 = 2n ** 128n;
+
+    if (data >= dataMaxUint8) {
+      valid = false;
+      this.logger.error(`[PriceDataOverflowChecker] data: got ${data}, max uint8 ${dataMaxUint8}`);
+    }
+
+    if (heartbeat >= heartbeatMaxUint24) {
+      valid = false;
+      this.logger.error(`[PriceDataOverflowChecker] heartbeat: got ${heartbeat}, max uint8 ${heartbeatMaxUint24}`);
```
Comment (leossantiago) [inline src/services/deviationsFeeds/PriceDataOverflowChecker.ts:27]: Typo on the following three logs. Shouldn't be uint24, uint32 and uint128?

**Review by leossantiago — APPROVED** [review]

File: src/services/deviationsFeeds/PriceDataOverflowChecker.ts (hunk 3 above)
Comment (DZariusz) [reply src/services/deviationsFeeds/PriceDataOverflowChecker.ts]: done

**Pull request merged**